### DISCUSSION TITLE
Looking for security arguments/tips for Vaultwarden

### DISCUSSION THREAD

**Discussion by opsora-labs** [dani-garcia/vaultwarden#5042] [Q&A]
Hi,
I recommended a customer to implement Vaultwarden.
From the Bitwarden documentation I know it using End-To-End encryption with the encryption keys not being saved on servers, so even if someone steals the database, they won't be able to decrypt it.

Does someone has/know here an article that describes the encryption and what to do to improve the security?
Thanks!
Tobias

**Comment by stefan0xC** [comment]
Do you mean something like the [Bitwarden Security Whitepaper](https://bitwarden.com/help/bitwarden-security-white-paper/)?

**Reply by opsora-labs** [reply]
Yes something like this. I already checked the document and send the information to the customer. But we are not experts in that field.
For the business decision it would be perfect if there would be an easy-to-read article how it is safe, and how the data can't be stolen.
Instead of gathering the information myself I hoped that someone has something saved here :-)
